src/ontodia/diagram/halo/haloMarkup.tsx

```diff
@@ -0,0 +1,49 @@
+import * as joint from 'jointjs';
+import * as React from 'react';
+
+export interface Props {
+    cellView: joint.dia.CellView;
+    onDelete: () => void;
+    onExpand: () => void;
+}
+
+export interface State {
+    cellIsExpanded: boolean;
+}
+
+export class HaloMarkup extends React.Component<Props, State> {
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            cellIsExpanded: this.props.cellView.model.get('isExpanded'),
+        };
+    }
+
+    public componentDidMount() {
+        this.props.cellView.model.on('change:isExpanded', this.setCellIsExpanded);
+    }
+
+    public componentWillUnmount() {
+        this.props.cellView.model.off('change:isExpanded', this.setCellIsExpanded);
+    }
+
+    private setCellIsExpanded = () => {
+        this.setState({
+            cellIsExpanded: this.props.cellView.model.get('isExpanded'),
+        });
+    };
+
+    render() {
+        let cellIsExpanded = this.state.cellIsExpanded;
+
+        return (
+            <div>
+                <div className='ontodia-halo__btn ontodia-halo__btn--delete' onClick={this.props.onDelete}/>
```

Comment:
This is not a right way to use BEM. `ontodia-halo__btn--delete` sounds like a `delete` modifier on a `ontodia-halo__btn` element of `ontodia-halo` component.
Instead there are should be `ontodia-halo__delete-button` (or just `ontodia-halo__delete`) and `ontodia-halo__expand` with `ontodia-halo__expand--closed` and `ontodia-halo__expand--open` modifiers.